### PR TITLE
Projects Schema, Get endpoints, and JWT Authentication

### PR DESCRIPTION
**Pull request by ahmad-diaa**
In this PR:
- Projects datamodel
- Projects get endpoints
- Middleware to authenticate JWT tokens
- moving configuration variables to env file

Part of #12

### CODE REVIEW

File: backend-zs/middleware/auth.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const jwt = require("jsonwebtoken");
+
+module.exports = {
+    authenticateUser: function (req, res, next) {
```
Comment (ahmad-diaa) [inline backend-zs/middleware/auth.js]: @dimitrityan I added the middleware to decode the jwt token for endpoints that would only be accessed by logged in users. Decoded info will be found at req.user. Any comments on the implementation? Is this how you are planning to do it?

File: backend-zs/models/project.model.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        required: true
+    },
+    coordinates: {
+        type: [Number],
+        required: true
+    }
+});
+
+
+var Project = mongoose.Schema({
```
Comment (klarmannsonia) [inline backend-zs/models/project.model.js:17]: The definition of a mongoose Schema requires instantiation with 'new'.

Comment (ahmad-diaa) [reply backend-zs/models/project.model.js:17]: Both work the same way (https://stackoverflow.com/questions/39719454/creating-mongoose-schemas-with-or-without-new-keyword). I wanted to be consistent with the implementation of the user model.

Comment (daliamaarek) [reply backend-zs/models/project.model.js:17]: In that case, I guess we should remove ``new`` from the Point instantiation

**Review by klarmannsonia — APPROVED** [review]

**Review by daliamaarek — APPROVED** [review]

**Pull request merged**